Bump to upstream

SimoneN64 opened the pull request:
Finally was able to do it properly. I hope I didn't miss anything. It compiles and runs properly for me, but people more experienced in the TAS world can tell me. Please review

Tilka committed:
AX: add support for biquad filtering

This fixes e.g. the overly loud wind in "I SPY: Spooky Mansion".

Tilka committed:
AX: enable low-pass/biquad filtering of Wiimote audio

Geotale committed:
Proper Subnormal Rounding When Interpreting

During 25-bit rounding, subnormals are "normalized"
This would normally mean that the exponent needs to be able to be <-1023
Instead, you can modify at what bit you round and get the same results!
This is done by finding the highest bit and shifting right the round bit

Co-Authored-By: JosJuice <josjuice@gmail.com>

mitaclaw committed:
BitUtils: Constexpr BitCastToArray, Remove BitCastFromArray

`std::bit_cast` participates in overload resolution only if `sizeof(To) == sizeof(From)` and both `To` and `From` are *TriviallyCopyable* types, so the static assertions here can be removed. `[[nodiscard]]` was added as well.

JosJuice committed:
Partially revert "Revert "Audit uses of IsRunning and GetState""

This reverts the revert commit bc67fc97c39628c76a4dbca411b0e8a9bfaf726a,
except for the changes in BaseConfigLoader.cpp, which caused the bug
that made us revert 72cf2bdb87f09deff22e1085de3290126aa4ad05. PR 12917
contains an improved change to BaseConfigLoader.cpp, which can be merged
(or rejected) independently.

A few changes have also been made based on review comments.

nivomi committed:
Add instructive text to disabled RetroAchievements login button

Changes the RetroAchievements "Log In" button's text to "To log in, stop the current emulation." when the button is disabled because an emulation session is active. This allows a user to understand why the button is disabled, and how this state can be resolved.

Previously, it could be unclear why this button was disabled without an understanding of the underlying system.

Co-Authored-By: JosJuice <josjuice@gmail.com>

LillyJadeKatrin committed:
Properly Notify when Achievements Login Fails

mitaclaw committed:
BranchWatchDialog: De-lambda-ize Constructor

Dentomologist committed:
Extract ARCodeWidget/GeckoCodeWidget code loading to functions

Dentomologist committed:
CheatsManager: Create ARCodeWidget and GeckoCodeWidget only once.

Create ARCodeWidget and GeckoCodeWidget once on startup rather than
every time a game is launched or shutdown.

In addition to losing focus on the tab (since the previous widget and
tab no longer existed), the behavior prior to this commit could cause a
crash if the user initiated a game shutdown and then opened a code edit
window since the AR/GeckoCodeWidget would get deleted in the meantime.

mitaclaw committed:
Branch Watch Tool: Toolbar Visiblity Menu

Adds the ability to hide unneeded features of the controls toolbar.

mitaclaw committed:
BranchWatchDialog: Defer Layout Construction When Possible

The main layout, tool controls layout, and misc. controls layout all don't need the QLayout constructed so early.

mitaclaw committed:
BranchWatchDialog: Refactor For LoadQSettings / SaveQSettings

mitaclaw committed:
BranchWatchDialog: Disconnect Slots When Hidden

mitaclaw committed:
BranchWatchDialog: Clean Up Object Parenting and Prefer Auto

Objects which get parented automatically by later processing now pass a nullptr to the constructor to make the intent clearer. Also fixed "true" and "false" not being translatable strings.

JosJuice committed:
DolphinQt: Simplify the Speed Limit description

The description of the Speed Limit setting currently uses a lot of
complicated terms, like "emulated time" (known to many Dolphin
developers, but in my experience not known by even advanced emulator
users) and "maximum time scale" (I have never heard it before). The
meaning of "sustainable" is also unclear in context.

This commit rewords the description to be easier to understand.

mitaclaw committed:
BranchWatchDialog: Refactor Context Menus

Instead of one wildly complex context menu constructed lazily, now three manageable context menus are constructed proactively.

mitaclaw committed:
BranchWatch: Don't Save Irrelevant Hits In Reduction Phase

mitaclaw committed:
Branch Watch Tool: New Conditional Branch Inspection Tools

Invert conditions, invert decrement checks, and make conditional branches unconditional. USnapshotMetadata in prior versions of Dolphin is forward-compatible with these changes (tested on x86_64).

mitaclaw committed:
Branch Watch Tool: Ignore Apploader Branch Hits Concurrency Fix

Also removed worthless `Start` and `Pause` helpers from `Core::BranchWatch`.

Geotale committed:
Improve Integer Rounding Accuracy

Changes integer rounding to more closely meet the documentation
The documentation explains to round before doing any bounds checks
All this really does is make sure some exception bits won't be set wrong
This depends on the rounding mode, fixing cases such as:
- Round to even, (0x7fffffff, 0x7fffffff.8)
- Round to down, (0x7fffffff, 0x80000000)

This change also uses some standard functions for rounding
Previously using them was casting to an s32 directly, now keeps the f64
RoundToIntegerMode introduced due to roundeven not being part of C++17

Finally, it can change a >0x7fffffff to >=0x80000000, done because:
- It looks nicer now with integers (I liked 0s)
- It gives ever so slightly better codegen on Aarch64

Co-Authored-By: JosJuice <josjuice@gmail.com>

mitaclaw committed:
BranchWatchDialog: Listen For `clicked` Signal Rather Than `pressed`

The latter signal was used by mistake, see PR #8263.

parona-source committed:
Add support for libfmt-11

fmt::join was moved into fmt/ranges.h

Signed-off-by: Alfred Wingate <parona@protonmail.com>

Tilka committed:
Small wording changes for dump conversions

Tilka committed:
DolphinTool: print title IDs in hex

Tilka committed:
VideoCommon: force 32-byte alignment for display list address and size

Tilka committed:
GameSettings: remove patches for "Wallace & Gromit in Project Zoo"

These patches were added in bde9a459cdd7a465cf9c9e275bd4562bb072ae7e
and enabled by default in 36ecfdd6b597b0264625d06320a5308147bbc55b.
They allowed the game to run in Dolphin but disabled dynamic shadows.
The game adds 4 to the start address of otherwise valid display lists
for no obvious reason. Now that Dolphin forces 32-byte alignment these
patches are no longer needed.

BryanJacobs committed:
Graphics: Adapt aspect ratio when SBS/TAB 3D is used

Adds support for choosing to present the full resolution
independently to each eye when using side-by-side or
top-and-bottom 3D.

JosJuice committed:
MSBuild: Use SkipUnchangedFiles with Copy

Just a bit of simplification.

JosJuice committed:
DolphinQt: Delete output Sys folder before copying to it

For a long time now, we've had a problem where game INIs persist in
the copied Sys folder if they've been deleted from the original Sys
folder. (I still have hundreds of game INIs locally that only set
EmulationStateId, and we removed those game INIs 6 years ago. On the
buildbot, we do occasionally clear out the build directories manually,
so I'd assume it's not quite as bad there.)

This commit fixes the problem by deleting the output Sys folder before
copying the original Sys folder to the output Sys folder. This should be
a bit slower, but in my testing, the difference seems small. At least if
you have an SSD, which I really hope people have nowadays!

Operating systems other than Windows have not been touched, because:

* Android: Already explicitly deletes the output Sys folder.
* macOS: Does some magic to put the Sys folder in the app bundle, which I
  will simply assume isn't affected by this problem, without testing.
* Linux: Expects the person building to manually manage the Sys folder.

JosJuice committed:
UnitTests: Delete output Sys folder before copying to it

Like the previous commit, but for UnitTests. This time all operating
systems were affected.

I also made UnitTests.vcxproj use the same way of copying as
DolphinQt.vcxproj, just for consistency.

duckymomo360 committed:
GameINI: Fix tattoo texture saving in Rabbids Go Home

suca333 committed:
fix fmt folder for Scripting project

suca333 committed:
add API/ folder to VS compilation props

OatmealDome committed:
Merge pull request #10556 from cpba/detectflatpak

Detect when running inside a flatpak sandbox

Tilka committed:
Merge pull request #13012 from mitaclaw/Use-contains-method-2

Use contains method 2

Commit:
Merge branch 'dolphin-emu:master' into master

Tilka committed:
Merge pull request #13017 from Tilka/foo

DolphinTool: print title IDs in hex

Tilka committed:
Merge pull request #13010 from OatmealDome/goodbye-steam

Remove Steam support

Tilka committed:
Merge pull request #13015 from tygyh/Use-boolean-literals

Replace boolean constants with boolean literals

Commit:
Merge branch 'dolphin-emu:master' into master

Tilka committed:
Merge pull request #13011 from JosJuice/clear-sys

Delete output Sys folder before copying to it

Tilka committed:
Merge pull request #12889 from mitaclaw/constexpr-bitutils-1

BitUtils: Constexpr BitCastToArray, Remove BitCastFromArray

Tilka committed:
Merge pull request #13008 from noahpistilli/kd-get-time-triggers

IOS/KD: Implement Get Time Triggers

Commit:
Merge branch 'dolphin-emu:master' into master

JosJuice committed:
UnitTests: Remove MSBuild input/output tracking

Input/output tracking is intended to speed up incremental builds by
skipping a target when it doesn't need to be built. However, this seems
to be unreliable for UnitTests' AfterBuild target. Let's remove it.

It's still the case that UnitTests' AfterBuild target will be skipped if
UnitTests doesn't need to be rebuilt. (Note that UnitTests always needs
to be rebuilt if SCMRevGen changed.)

I haven't seen these problems with DolphinQt's rather similar
input/output tracking. This may be because DolphinQt's one also has the
exe file as an input/output.

Tilka committed:
Merge pull request #13019 from JosJuice/clear-sys-2

UnitTests: Remove MSBuild input/output tracking

Commit:
Merge branch 'dolphin-emu:master' into master

Tilka committed:
Merge pull request #12773 from Tilka/zelda_filters

DSPHLE/Zelda: Add two missing filters

Pokechu22 committed:
Fix postprocessing shader not changing when setting stereo mode

Fixes https://bugs.dolphin-emu.org/issues/13593. Before, it would attempt to
use the old shader, which did not exist for the new stereo mode. Changing the
postprocessing shader afterwards would work properly, although passive 3D only
has one option, so it was just broken without restarting Dolphin.

This also fixes the UI not updating when using one of the stereo toggle hotkeys.

Tilka committed:
Merge pull request #13005 from Tilka/display_list

VideoCommon: force display list alignment

Tilka committed:
Merge pull request #13021 from Pokechu22/stereo-post-processing-ui

Fix postprocessing shader not changing when setting stereo mode

AdmiralCurtiss committed:
Merge pull request #12144 from khg8m3r/sdl2-update

Externals: Update SDL to 2.30.6

Tilka committed:
Merge pull request #12863 from JosJuice/android-gamepad-sensors

Android: Fix and enable input device sensor input

Tilka committed:
AX: add analytics quirks for wiimote filters

I'm fairly sure the implementation is correct but I've not been able to
find a game that actually makes use of these filters.

tygyh committed:
Remove redundant semicolons

Tilka committed:
Merge pull request #12974 from JosJuice/simplify-speed-limit-description

DolphinQt: Simplify the Speed Limit description

AdmiralCurtiss committed:
Merge pull request #12942 from nivomi/master

RetroAchievements: Add instructive text to disabled login button

Commit:
Merge branch 'dolphin-emu:master' into master

JosJuice committed:
Translation resources sync with Transifex

Tilka committed:
Merge pull request #13022 from tygyh/Remove-redundant-semicolons

Remove redundant semicolons

lycheefox committed:
Skylanders: Add Mobile Hot Streak and fix typo

Commit:
Merge branch 'dolphin-emu:master' into master

Blounard commented:
I managed to build it, dolphin behave nicely, but the scripting system seems to have an issue.
While the scripts actually work, they seem to not trigger the events. This means a script which is supposed to run a function on every frame won't actually run this function correctly. For example, the infodisplay doesn't work.

OatmealDome committed:
Merge pull request #13028 from lycheefox/skylanders

Skylanders: Add Mobile Hot Streak and fix typo

JosJuice committed:
DolphinQt: Fix "Sddress" typo

OatmealDome committed:
Merge pull request #13029 from JosJuice/sddress

DolphinQt: Fix "Sddress" typo

Commit:
Merge branch 'dolphin-emu:master' into master

mitaclaw committed:
TypeUtils: Remove Common::Fill

This temporary solution is no longer needed.

lycheefox committed:
Skylanders: Fix Hot Head (Sparkle)

mitaclaw committed:
Modernize `std::adjacent_find` with ranges

mitaclaw committed:
Simplify `std::fill` with `std::fill_n`

Tilka committed:
Merge pull request #13034 from lycheefox/skylanders

Skylanders: Fix Hot Head (Sparkle)

Tilka committed:
Merge pull request #13033 from mitaclaw/remove-common-fill

TypeUtils: Remove Common::Fill

Commit:
Merge branch 'dolphin-emu:master' into master

JosJuice committed:
Merge pull request #13009 from BryanJacobs/master

Graphics: Adapt aspect ratio when SBS/TAB 3D is used

Tilka committed:
Merge pull request #10663 from Tilka/ax_biquad

AX: add support for biquad filtering

PatrickFerry committed:
Gameini: Enable PAL60 for "The Last Story"

PatrickFerry committed:
Gameini: Enable PAL60 for "Boom Street"

JosJuice committed:
Translation resources sync with Transifex

Commit:
Merge branch 'dolphin-emu:master' into master

Commit:
Merge branch 'dolphin-emu:master' into master

PatrickFerry committed:
Gameini: Enable PAL60 for "Project Zero II: Wii Edition"

suca333 committed:
Things I had missed

Dentomologist committed:
ARWidget: Disable Edit and Remove buttons when no code is selected

Also some minor refactoring of nearby/related code:
* Make non-obvious variable types explicit instead of auto.
* Throw some consts around.
* Use setDisabled(empty) instead of setEnabled(!empty).

Dentomologist committed:
ARCodeWidget: Remove unnecessary call to OnSelectionChanged

Before the call to OnSelectionChange, m_code_edit and m_code_remove are
disabled and UpdateList calls m_code_list->clear(), thereby deselecting
any selected items.

When no items are selected, OnSelectionChange disables m_code_edit and
m_code_remove and then returns. Since that was already done, the call
doesn't change anything and can be removed.

ghost commented:
> I managed to build it, dolphin behave nicely, but the scripting system seems to have an issue. While the scripts actually work, they seem to not trigger the events. This means a script which is supposed to run a function on every frame won't actually run this function correctly. For example, the infodisplay doesn't work.

Should be fixed. I tested a script that prints "frame advance" on every `@event.on_frameadvance()` and it works

Dentomologist committed:
HotkeyScheduler: Handle Wii Remote connections independently

Allow connecting or disconnecting multiple Wii Remotes simultaneously
instead of only handling the highest index whose hotkey is pressed. This
allows using a single hotkey to toggle multiple remotes.

Ferdi265 committed:
RVZ: Fix undefined behaviour when copying 0 bytes to a null pointer

A vector of length 0 can have a null data pointer, which causes UB when
passed to memcpy, so only copy when we actually have data to copy. This
caused crashes in certain cases when compiling Dolphin with Clang and
LTO enabled.

Tilka committed:
Merge pull request #13037 from Dentomologist/hotkeyscheduler_handle_wiimote_connections_independently

HotkeyScheduler: Handle Wii Remote connections independently

Commit:
Merge branch 'dolphin-emu:master' into master

Dentomologist committed:
MainWindow: Prevent Confirm On Stop dialog from being hidden

Set the Render Window as the parent of the Confirm On Stop confirmation
dialog when Keep Window On Top is enabled, ensuring it will always be
visible.

Previously, when Confirm On Stop and Keep Window On Top were both
enabled the Confirm On Stop dialog could be hidden by the render window
in the following situations:
* Clicking Stop in the Main Window
* Clicking the Main Window's close button
* Pressing the Stop hotkey while in FullScreen mode

This was particularly troublesome because the confirm dialog is modal,
preventing the user from moving the render window out of the way if it
was obscuring the dialog.

Fixes https://bugs.dolphin-emu.org/issues/13247.

Tilka committed:
Merge pull request #13036 from PatrickFerry/gameini_enable_pal60

Gameini: Enable PAL60 for Some PAL Games

Commit:
vulkan: Add line number to vulkan error logger

This is a minor improvement to add line numbers to the LOG_VULKAN_ERROR
define. Basically error logs for Vulkan will now look like:

```
// This
25:03:347 VideoBackends/Vulkan/VulkanLoader.cpp:247 E[Video]: (WaitForCommandBufferCompletion:278) vkWaitForFences failed:  (2: VK_TIMEOUT)

// Instead of
15:45:154 VideoBackends/Vulkan/VulkanLoader.cpp:247 E[Video]: (WaitForCommandBufferCompletion) vkWaitForFences failed:  (2: VK_TIMEOUT)
```

Commit:
Merge branch 'dolphin-emu:master' into master

Tilka committed:
GameSettings: force EFB-to-RAM for My First Songs

Tilka committed:
Merge pull request #13038 from Ferdi265/fix-rvz-clang-ub

RVZ: Fix undefined behaviour when copying 0 bytes to a null pointer

Tilka committed:
Merge pull request #13039 from Dentomologist/mainwindow_fix_hidden_confirm_on_stop_dialog

MainWindow: Prevent Confirm On Stop dialog from being hidden by the Render window

Tilka committed:
Merge pull request #13043 from Tilka/my_first_songs

GameSettings: force EFB-to-RAM for My First Songs

Tilka committed:
Merge pull request #13042 from homeisfar/vk_improve_logger

vulkan: Add line number to vulkan error logger

AdmiralCurtiss committed:
Merge pull request #13031 from parona-source/libfmt-11

Add support for libfmt-11

JosJuice committed:
Interpreter: Fix subfic carry calculation

This was accidentally using the instruction's output instead of the
instruction's input when the input and output registers were the same.

Tilka committed:
Merge pull request #13044 from JosJuice/subfic-carry

Interpreter: Fix subfic carry calculation

mitaclaw committed:
BranchWatchDialog: Const Correctness

m_index_list_temp should not be imagined as a member of `BranchWatchDialog`, so it is now mutable to allow for more const member functions.

mitaclaw committed:
BranchWatchDialog: Improve Branch Was/Not Overwritten Buttons

Giving a warning if these are used when not usable is bad UX.

JosJuice committed:
Translation resources sync with Transifex

Commit:
Merge branch 'dolphin-emu:master' into master

mitaclaw committed:
CachedInterpreterEmitter: Fix `std::memcpy` UB

I wasn't aware that even with a size of zero, it's still not safe to pass a nullptr to `std::memcpy`. When `CachedInterpreterEmitter::PoisonCallback` is written, UB is happening.

AdmiralCurtiss committed:
Merge pull request #13047 from mitaclaw/cached-interpreter-memcpy-ub

CachedInterpreterEmitter: Fix `std::memcpy` UB

PatrickFerry committed:
GameINI: Remove Broken Cheat from "Paper Mario: The Thousand Year Door"

This is causing graphical glitches on some sprites, specifically the partner character after loading a game.

TellowKrinkle committed:
VideoBackends:Vulkan: Consolidate feature checking into one struct

OatmealDome committed:
ScmRevGen: Bump major version to 2409

TellowKrinkle committed:
VideoBackends:Vulkan: Use Vulkan 1.2 driverID to detect MoltenVK

Previously we'd assume all Apple GPUs were MoltenVK, including those running on Asahi Linux with open source Honeykrisp drivers.

Commit:
Merge branch 'dolphin-emu:master' into master

mitaclaw committed:
SI_Device: Remove Dead Config Code

OatmealDome committed:
Merge branch 'release-prep-2409'

Tilka committed:
DSPHLE/AX: fix low-pass/biquad clipping

The low-pass and biquad filters run in set40 mode where accessing ac#.m
returns the value of ac#.hm clamped to 16 bits.

This fixes the crackling in "Need for Speed: Nitro" (issue 13610).

Also make the lower bound match hardware (-0x8000 instead of -0x7FFF).

JMC47 committed:
Merge pull request #12977 from mitaclaw/branch-watch-tool-fixes-4

Branch Watch Tool: Refactors, Fixes, and Features

Dentomologist committed:
Texture Dumping: Show OSD message on startup and when toggled

If texture dumping is enabled, notify the user on emulation startup
using an On Screen Display message.

Also notify the user when texture dumping is toggled.

Addresses https://bugs.dolphin-emu.org/issues/12445.

Commit:
Merge branch 'dolphin-emu:master' into master

Tilka committed:
Merge pull request #13052 from Tilka/ax

DSPHLE/AX: fix low-pass/biquad clipping

JosJuice committed:
Merge pull request #13051 from mitaclaw/dead-config-code-2

SI_Device: Remove Dead Config Code

JMC47 committed:
Merge pull request #12966 from Dentomologist/cheatsmanager_create_code_widgets_only_once

CheatsManager: Create Action Replay and Gecko code widgets only once

JosJuice committed:
Merge pull request #12981 from Geotale/proper-integer-rounding

Improve Integer Rounding Accuracy

JosJuice committed:
DiscIO: Fix a typo in an RVZPack comment

JosJuice committed:
Merge pull request #12813 from Geotale/interpreter-subnormal-rounding

Proper Subnormal Rounding When Interpreting

Tilka committed:
Merge pull request #13057 from JosJuice/rvzpack-typo

DiscIO: Fix a typo in an RVZPack comment

Dentomologist committed:
GameSettings: Disable GPU Texture Decoding when needed

Arbitrary Mipmap Detection doesn't work when GPU Texture Decoding is
enabled, so disable GPU Texture Decoding for games where the .ini
enables Arbitrary Mipmap Detection.

Co-authored-by: jeremyie <73066289+McAchi@users.noreply.github.com>

Dentomologist committed:
Graphics: Clarify relationship between incompatible settings

Explicitly state that Manual Texture Sampling disables Anisotropic
Filtering and that GPU Texture Decoding disables Arbitrary Mipmap
Detection.

Commit:
Merge branch 'dolphin-emu:master' into master

retroNUC committed:
Add generation of rcheevos hash to DolphinTool

Fixed coding standards issue

CalculateHash return string instead of using char array param

PatrickFerry committed:
GameINI: Disable ICache for Happy Feet

The DisableICache setting is supposed to be for [Core]

PatrickFerry committed:
GameINI: Enable Vertex Rounding for "Teenage Mutant Ninja Turtles 2: Battle Nexus"

ImmediateXFBEnable was disabled originally when enabling VertexRounding was intended

https://bugs.dolphin-emu.org/issues/10186

PatrickFerry committed:
GameINI: Enable Vertex Rounding for "Teenage Mutant Ninja Turtles 3: Mutant Nightmare"

ImmediateXFBEnable was disabled originally when enabling VertexRounding was intended

https://bugs.dolphin-emu.org/issues/10186

PatrickFerry committed:
GameINI: Disable Broken Enhancements for Solvalou

The wrong filename was used so the settings weren't being applied.

The graphical issues reported were from enhancements that create graphical issues.

EFBToTextureEnable solves some graphical issues but those issues appear for only a few frames, the bulk of the game does not benefit while the setting while it is very sore on performance.

Tilka committed:
Merge pull request #13035 from retroNUC/cheevo-hash

Add generation of rcheevos hash as an option in DolphinTool

suca333 committed:
Merge branch 'master' of https://github.com/dolphin-emu/dolphin into dolphin-emu-master

# Conflicts:
#	Readme.md

suca333 committed:
Merge branch 'dolphin-emu-master'

Tilka committed:
Merge pull request #13064 from PatrickFerry/gameini_fixing_mistakes

GameINI: Fixing Various Mistakes

Commit:
Merge branch 'dolphin-emu:master' into master

mitaclaw committed:
DolphinQt: A Ubiquitous Signal For When Breakpoints Change

There were three distinct mechanisms for signaling breakpoint changes in DolphinQt, and the wiring had room for improvement. The behavior of these signals has been consolidated into the new `Host::PPCBreakpointsChanged` signal, which can be emitted from anywhere in DolphinQt to properly update breakpoints everywhere in DolphinQt.

This improves a few things:
- For the `CodeViewWidget` and `MemoryViewWidget`, signals no longer need to propagate through the `CodeWidget` and `MemoryWidget` (respectively) to reach their destination (incoming or outgoing).
- For the `BreakpointWidget`, by self-triggering from its own signal, it no longer must manually call `Update()` after all of the emission sites.
- For the `BranchWatchDialog`, it now has one less thing it must go through the `CodeWidget` for, which is a plus.

mitaclaw committed:
BreakpointWidget: Update when `Host::PPCSymbolsChanged`

Otherwise, outdated symbol names will remain.

mitaclaw committed:
Modernize `std::find` with ranges

Tilka committed:
Sw/Rasterizer: fix two elder typos, AD 2009

JMC47 committed:
Merge pull request #13058 from Dentomologist/gamesettings_disable_gpu_texture_decoding_when_needed

GameSettings: Disable GPU Texture Decoding when needed

JMC47 committed:
Merge pull request #13053 from Dentomologist/texture_dumping_show_osd_on_startup_and_toggle

Texture Dumping: Show OSD message on startup and when toggled

Tilka committed:
Sw/Tev: drop unused macro

mitaclaw committed:
GeckoCode: Remove Dead Code

mitaclaw committed:
Modernize `std::find_if` with ranges

In BTEmu.cpp, `std::mem_fn` was not necessary for the predicate to compile.

Tilka committed:
Merge pull request #13067 from mitaclaw/dead-code-3

GeckoCode: Remove Dead Code

mitaclaw committed:
IniFile: Migrate `Common::CaseInsensitiveLess` to StringUtil

Migrating `Common::CaseInsensitiveLess` to StringUtil.h will hopefully discourage rolling one's own solution in the future for case-insensitive associative containers when this (quite robust!) solution already exists.

`Common::CaseInsensitiveStringCompare::IsEqual` was removed in favor of using the `Common::CaseInsensitiveEquals` function.

The `a.size() != b.size()` condition in `Common::CaseInsensitiveEquals` can be removed, since `std::ranges::equal` already checks this condition (confirmed in libc++).

JosJuice committed:
Merge pull request #13013 from duckymomo360/rgh-tattoos-fix

GameINI: Fix tattoo texture saving in Rabbids Go Home

mitaclaw committed:
StringUtil: Remove `JoinStrings`

With 12 uses of `JoinStrings` in the codebase vs 36 uses of `fmt::join`, fmtlib's range adapter for string concatenation with delimiters is clearly the preferred option.

Commit:
Merge branch 'dolphin-emu:master' into master

JosJuice committed:
Android: Add the "Real Balance Board" setting

robxnano committed:
linux: Use D-Bus to inhibit screensaver

Pokechu22 committed:
Fix random crash when using the cheat search

Before, Dolphin would randomly crash when updating the cheat search when automatic refresh was enabled. (Having a large number of addresses listed, e.g. by starting with an "any value" search, may contribute). The crash was due to QTableWidget::item returning nullptr in RefreshGUICurrentValues, presumably due to the table being resized on the UI thread while the emulated CPU thread was updating the values. I've fixed this by pausing the CPU thread for the entirety of OnNextScanClicked; this eliminated crashes in my testing.

Dentomologist committed:
GraphicsWindow: Fix crash when opening during emulation startup

Fix a crash when opening the Graphics window for the first time during
emulation startup when the backend is Vulkan, D3D11, or D3D12.

Don't call PopulateBackendInfo() from the Host thread when the core is
starting up. First, the function has already been called in Core::Init()
so we don't need to again. More importantly, PopulateBackendInfo() calls
g_video_backend->InitBackendInfo(), and the Vulkan and D3D
implementations of those functions load and then unload libraries (and
their associated function pointers) which are potentially in use by
other threads.

This crash was reliably reproducible with the following steps:
1) Select an affected backend.
2) Enable "Compile Shaders Before Starting"
3) Delete the cached shaders (but not the .uidcache file) for the game
   you're testing.
4) Close and reopen Dolphin.
5) Start the game.
6) While the game is still booting or compiling shaders, open the
   Graphics window for the first time in that Dolphin session.

Fixes https://bugs.dolphin-emu.org/issues/13634.

Pokechu22 committed:
Cache normals in addition to binormals and tangents

Fixes LIT (https://bugs.dolphin-emu.org/issues/13635). The text does not include normals, but has lighting enabled. With the previous default of (0, 0, 0), lighting was always black (as dot(X, (0, 0, 0)) is always 0). It seems like the normal from the map in the background (0, 0, 1) is re-used.

LIT also has the vertex color enabled while vertex color is not specified, the same as SMS's debug cubes; the default MissingColorValue GameINI value of solid white seems to work correctly in this case.

Tilka committed:
Merge pull request #13072 from Tilka/videosw

VideoSoftware: small cleanup

JMC47 committed:
Merge pull request #13049 from PatrickFerry/gameini_remove_broken_cheat

GameINI: Remove Broken Cheat from "Paper Mario: The Thousand Year Door"

JMC47 committed:
Merge pull request #13071 from Pokechu22/cheat-search-crash

Fix random crash when using the cheat search

robxnano committed:
DolphinQt: Remove unneeded Win32 code

Dolphin already uses QStyleHints::colorScheme to detect dark mode,
so this code is now redundant.

Dentomologist committed:
Core: Remove redundant call to PopulateBackendInfo

Remove the second of two calls to PopulateBackendInfo during emulation
startup. This call was originally the only one, but b214e0e added an
earlier call to handle the backend being changed by the GameINI.

The PR discussion doesn't explain why the original call was left in; I
suspect it was just overlooked.

As a bonus, this removes one of the extra copies of the "Video Info" On
Screen Display message at startup when using OpenGL.

Dentomologist committed:
VideoBackendBase: Check Core state in PopulateBackendInfo

Remove the PopulateBackendInfoFromUI function, which had a single caller
(GraphicsWindow::OnBackendChanged) and checked that the core wasn't
running or starting before calling PopulateBackendInfo.

Move the core state check into PopulateBackendInfo and have
OnBackendChanged call that instead. This guarantees the check is
performed by all callers of PopulateBackendInfo, preventing
potential reintroduction of the crash fixed in 3d4ae63f if another call
to PopulateBackendInfo is added.

As of the previous commit the only other caller of PopulateBackendInfo
is Core::Init shortly before s_state is set to Starting, so it will
always pass the check and so maintain its current behavior.

JMC47 committed:
Merge pull request #13070 from mitaclaw/remove-join-strings

StringUtil: Remove `JoinStrings`

JMC47 committed:
Merge pull request #13065 from mitaclaw/ppc-breakpoints-signal

DolphinQt: A Ubiquitous Signal For When Breakpoints Change

JMC47 committed:
Merge pull request #13068 from mitaclaw/redundant-case-insensitive

IniFile: Migrate `Common::CaseInsensitiveLess` to StringUtil

Tilka committed:
Merge pull request #13077 from Dentomologist/graphicswindow_fix_first_opening_during_emulation_startup_crash

GraphicsWindow: Fix crash when opening for the first time during emulation startup

Tilka committed:
Merge pull request #13062 from Dentomologist/graphics_clarify_relationship_between_incompatible_settings

Graphics: Clarify relationship between incompatible settings

Dentomologist committed:
VideoBackendBase: Only populate backend info when uninitialized

Prevent potential issues when creating the Graphics window (and thus
calling PopulateBackendInfo) while the core state is Stopping, like we
already do while it's Starting or Running.

JosJuice committed:
Android: Add 16 KiB page size support

So far, Dolphin hasn't been able to run on Android devices that use a
16 KiB page size. 16 KiB page sizes are a very new Android feature that
no phones have shipped with so far, so we're still compatible with the
phones that exist out there, but let's get this fixed before phones
start shipping with 16 KiB page sizes.

Because I couldn't get Android Studio's emulator to work, I haven't been
able to confirm that this change actually makes Dolphin fully compatible
with devices that use a 16 KiB page size. But I have confirmed that this
doesn't break anything on a regular 4 KiB page size device.

JosJuice committed:
Android: Use enableEdgeToEdge

Dolphin has been using edge-to-edge rendering for a little while now,
but it has required a bit of manual work. Now that edge-to-edge is
becoming something expected of apps in Android 15, there's a nicer API
we can use.

Tested on Android 8, 11 and 13, with no changes in behavior noted.

mitaclaw committed:
Simplify `std::copy` with `std::copy_n`

+ a surprise `std::memcpy` in VolumeVerifier.cpp.

Tilka committed:
Boot: write Apploader version into memory

When HLE-ing the apploader, this is necessary to avoid "Error #002"
errors. Homebrew doesn't care.

PatrickFerry committed:
GameINI: Fix Cheats for Conduit 2

These codes are actually Gecko codes so setting that correctly.

There were errors which pop up when running this game as the comments for AR codes need to be set with hash(#) instead of asterisk(*). I replaced those anyways.

Removed the url in the comment since it's inaccessible and not on wayback machine either.

Also corrected the GameID, the wrong one is listed.

PatrickFerry committed:
GameINI: Remove Obsolete Settings in Super Metroid Demo (Brawl VC)

Both UseXFB and UseRealXFB are obsolete since Hybrid XFB.

PatrickFerry committed:
GameINI: Remove Obsolete Setting in FaceBreaker K.O. Party

Hack is not a valid setting.

Also updated the GameID.

PatrickFerry committed:
GameINI: Rename INI for Naruto Collection (Demo)

The wrong filename was used so the settings weren't being applied.

(All it does is reset the Texture Cache Accuracy to the default setting.)

PatrickFerry committed:
GameINI: Disable AA in Sonic Heroes

The use of any anti-aliasing will cause graphical corruption, which occurs on the ground.

MaxAnisotropy is supposed to be for [Video_Enhancements], but also it doesn't cause any of the issues in game so it is being removed rather than set under the correct heading.

Added a comment on why these values are set.

PatrickFerry committed:
GameINI: Set Texture Cache Accuracy to "Safe" for UFC Personal Trainer: The Ultimate Fitness System

SafeTextureCacheColorSamples is supposed to be for [Video_Settings], but also the medium setting isn't enough to display the missing text.

mitaclaw committed:
VolumeVerifier: Promote `DEBUG_ASSERT` to `static_assert`

This runtime assertion was added before `std::is_sorted` became constexpr in the codebase.

JosJuice committed:
Merge pull request #13084 from mitaclaw/static-assert-promote

VolumeVerifier: Promote `DEBUG_ASSERT` to `static_assert`

Commit:
Merge branch 'dolphin-emu:master' into master

Tilka committed:
Merge pull request #13083 from Tilka/apploader

Boot: write Apploader version into memory

Tilka committed:
WII_IPC: rewrite multiply as 100_tbticks literal (NFC)

TellowKrinkle committed:
VideoBackends:Metal: Use ios_use_simdgroup_functions with SPIRV-Cross

It defaults to trying to emulate simdgroup functions on iOS

TellowKrinkle committed:
VideoCommon: Dirty pixel shader manager on efb scale changes

TellowKrinkle committed:
VideoBackends:Vulkan: Don't try to present if swapchain acquire failed

Tilka committed:
GameSettings: safe texture cache for "Winter Sports 3: The Great Tournament"

This is required to avoid glitches during Downhill Skiing.

TellowKrinkle committed:
VideoBackends:Vulkan: Wait until we've confirmed a successful AcquireNextImage before marking semaphores used

TellowKrinkle committed:
VideoBackends:Metal: subgroup_ops requires Apple7, not Apple6

SIMD-scoped permute operations are Apple6, but reduction operations are Apple7

JMC47 committed:
Merge pull request #13085 from TellowKrinkle/AutoSizeEFBScale

VideoCommon: Dirty pixel shader manager on efb scale changes

JMC47 committed:
Merge pull request #13081 from PatrickFerry/gameini_fixing_mistakes_part_2

GameINI: Fixing More Mistakes

Tilka committed:
Bluetooth: show vid:pid in error message

JMC47 committed:
Merge pull request #12953 from LillyJadeKatrin/retroachievements-fail-messaging

Achievements Fail Messaging

JMC47 committed:
Merge pull request #13087 from TellowKrinkle/PresentSkip

VideoBackends:Vulkan: Prevent freezes during window resize on Linux

JosJuice committed:
Merge pull request #13097 from Tilka/vidpid

Bluetooth: show vid:pid in error message

SuperSamus committed:
GameINI: Add Skylanders: Spyro's Adventure
`EFBEmulateFormatChanges = True` properly emulates the blur in the pause screen.
`ImmediateXFBEnable = False` to prevent epilepsy-inducing FMVs and main menu.

SuperSamus committed:
GameINI: Add Skylanders: Giants
`EFBEmulateFormatChanges = True` properly emulates the blur in the pause screen.
`ImmediateXFBEnable = False` to prevent epilepsy-inducing FMVs and main menu.

SuperSamus committed:
GameINI: Add Skylanders: Trap Team
`EFBEmulateFormatChanges = True` properly emulates the blur in the backgrounds.
`ImmediateXFBEnable = False` to prevent epilepsy-inducing FMVs.

JosJuice committed:
Merge pull request #13078 from Dentomologist/videobackendbase_only_populate_backend_info_when_uninitialized

VideoBackendBase: Only populate backend info when uninitialized

JMC47 committed:
Merge pull request #13098 from SuperSamus/skylanders-ini

GameINI: Add Skylanders titles

OatmealDome committed:
Merge pull request #13088 from TellowKrinkle/iOSFixes

VideoBackends:Metal: Fix bbox on newer iOS devices

JMC47 committed:
Merge pull request #13050 from TellowKrinkle/HKIsNotMVK

Vulkan: Don't do MoltenVK things on Asahi Linux

JosJuice committed:
Core: Include partition when calling GetGameTDBID for non-game partition

This problem was introduced in 8842a0f402ffe60a80ce7894bfd36afd039009fc
and is likely a copy-paste error.

Fixes the problem mentioned in the issue comment
https://bugs.dolphin-emu.org/issues/13640#note-2.

CrossVR committed:
Merge pull request #12208 from PatrickFerry/add-vertex-depth-range-case

Use Vertex Depth Range when zRange Exceeds farZ

Tilka committed:
Merge pull request #13101 from JosJuice/gametdbid-partition

Core: Include partition when calling GetGameTDBID for non-game partition

TellowKrinkle committed:
VideoBackends:Vulkan: Increase VMA Vulkan Version to 1.2

We now use Vulkan 1.2 if available

lycheefox committed:
Skylanders: Fix Head Rush (Nitro)

JMC47 committed:
Merge pull request #13103 from TellowKrinkle/VMAVulkanVersion

VideoBackends:Vulkan: Increase VMA Vulkan Version to 1.2

JMC47 committed:
Merge pull request #13105 from lycheefox/skylanders

Skylanders: Fix Head Rush (Nitro)

JMC47 committed:
Merge pull request #12918 from JosJuice/revert-revert-audit-get-state

Partially revert "Revert "Audit uses of IsRunning and GetState""

Pokechu22 committed:
FifoPlayer: Set up additional BATs in Wii mode

The light LIT fifolog from https://bugs.dolphin-emu.org/issues/13635 has position data at physical address 11ae3180. This works fine when using the memory viewer in physical mode, but the corresponding virtual address (91ae3180) previously didn't show anything in effective mode. It works fine now though.

This shouldn't affect playback of fifologs as everything in there uses physical addresses; this only impacts the memory viewer.

This logic was copied from CBoot::SetupBAT.

Pokechu22 committed:
Fifo analyzer: Fix indexed XF loads showing "bytes" instead of "words"

Indexed XF loads specify the number of 32-bit words (generally floats, but light data has some integers) to load, not the number of bytes. This was only a mistake in the fifo analyzer text; the actual implementation already loaded words.

Tilka committed:
Merge pull request #13110 from Pokechu22/fifo-analyzer-xf-index-words

Fifo analyzer: Fix indexed XF loads showing "bytes" instead of "words"

Tilka committed:
Merge pull request #13109 from Pokechu22/fifo-player-wii-bats

FifoPlayer: Set up additional BATs in Wii mode

Commit:
Merge branch 'dolphin-emu:master' into master

CrossVR committed:
Merge pull request #13086 from Tilka/gameini

GameSettings: safe texture cache for "Winter Sports 3: The Great Tournament"

mitaclaw committed:
C++20: Synthesize `operator!=` From `operator==`

The inequality operator is automatically generated by the compiler if `operator==` is defined.

NicholasWilsonDEV committed:
Update Contributing.md

fix typos

Tilka committed:
Merge pull request #13118 from BobConanDev/master

Update Contributing.md

SuperSamus committed:
Remove Bloom Definitions for Skylanders Trap Team

Sintendo committed:
JitArm64_RegCache: Add IsImm helper

Because checking if a register is a specific immediate is a rather
common operation.

Sintendo committed:
JitArm64_SystemRegisters: Optimize mtspr

No need to materialize the immediate if it is zero, we can just use WZR.

Before:
mov    w27, #0x0                 ; =0
str    w27, [x29, #0x1178]

After:
str    wzr, [x29, #0x1178]

Sintendo committed:
JitArm64_Integer: Use IsImm helper more

Commit:
Merge branch 'dolphin-emu:master' into master

Tilka committed:
Merge pull request #13121 from mitaclaw/synthesize-operator!=

C++20: Synthesize `operator!=` From `operator==`

Tilka committed:
Merge pull request #13119 from SuperSamus/skylanders-tt-gp-remove

Remove Bloom Definitions for Skylanders Trap Team

Tilka committed:
Merge pull request #13117 from mitaclaw/ranges-modernization-9-trivial-find

Ranges Algorithms Modernization - Find

Tilka committed:
Merge pull request #13095 from mitaclaw/ranges-modernization-6-n

Ranges Algorithms Modernization - N

Tilka committed:
Merge pull request #13120 from Sintendo/mtspr

JitArm64: Optimize mtspr

Tilka committed:
Merge pull request #13075 from robxnano/qt-color-scheme

Remove unneeded Win32 code for color scheme

Tilka committed:
Merge pull request #13076 from robxnano/dbus-inhibit

Linux: Use D-Bus to inhibit screensaver

jordan-woyak committed:
DolphinQt/MappingIndicator: Use Settings::IsThemeDark instead of reimplementing it.

jordan-woyak committed:
ControllerEmu: Change trigger threshold check to >= instead of > and set minimum threshold in UI to 1% to prevent user error.

Tilka committed:
Merge pull request #13123 from jordan-woyak/mapping-indicator-dark-theme-code-cleanup

DolphinQt/MappingIndicator: Use Settings::IsThemeDark instead of reimplementing it.

jordan-woyak committed:
DolphinQt: Auto hide the mapping window tab bar.

jordan-woyak committed:
DolphinQt: Restrict size of banner image in game properties info tab.

Tilka committed:
Merge pull request #13124 from jordan-woyak/trigger-threshold-99

ControllerEmu: Change trigger threshold check to prevent user error.

Tilka committed:
Merge pull request #13125 from jordan-woyak/mapping-tab-bar-auto-hide

DolphinQt: Auto hide the mapping window tab bar.

Tilka committed:
Merge pull request #13126 from jordan-woyak/no-big-banner

DolphinQt: Restrict size of banner image in game properties info tab.

Tilka committed:
Merge pull request #13080 from JosJuice/androidx-edge-to-edge

Android: Use enableEdgeToEdge

Tilka committed:
Merge pull request #13079 from JosJuice/android-16k

Android: Add 16 KiB page size support

Tilka committed:
Boot: don't error for NTSC IPL v1.2 of DOL-001 consoles

Checksum and descriptions from Redump.

Tilka committed:
Merge pull request #13074 from Pokechu22/normal-cache

Cache normals in addition to binormals and tangents

AGuy27 committed:
Create STZ.ini

AGuy27 committed:
Create S7E.ini

JosJuice committed:
Merge pull request #13127 from Tilka/dol001

Boot: don't error for NTSC IPL v1.2 of DOL-001 consoles

JMC47 committed:
Merge pull request #13128 from AGuy27/master

GameINI: Fix Transformers: Dark of the Moon hang on opening logos

JMC47 committed:
Merge pull request #13069 from JosJuice/android-real-balance-board

Android: Add the "Real Balance Board" setting

Commit:
Merge branch 'dolphin-emu:master' into master

ghost commented:
Closing, as I'm not willing to commit to my fork anymore. Nothing to do with the project, I love it, I just don't have much time anymore.
